### ISSUE TITLE
vscode format on save on big file cursor alway jump to the end of the file

### ISSUE THREAD

**Issue by Kyrosee** [prettier/prettier-vscode#2920]
Format on save on big js file(3700lines), cursor alway jump to the end of the file
prettier version 9.10.4
vscode 1.76.2

**Comment by madc0w** [comment]
I can confirm that this is a real issue, and it's driving me nuts.
The precise file size at which this behavior begins occurring is 100002 bytes. That is, formatting a file with 100001 bytes is fine, but adding one more byte breaks prettier. At least, this is true for a Javascript file. I've not tested other types.
I would include my example here, but it seems I'm only allowed to attach an image.

**Comment by foaudkajj** [comment]
+1 having the same problem.

**Comment by marekcambal** [comment]
yes, I can confirm the same issue
file being edited has ~3600 lines 

vscode 1.76.2
prettier version 9.10.4

setting `source.organizeImports` to false prevents cursor jumping to EOF
```json
"editor.codeActionsOnSave": {
    "source.organizeImports": false,
  },
```

**Comment by foaudkajj** [comment]
@marekcambal, please be aware that changing the setting only fixes the problem that happens while saving the file.
When the file is formatted with `alt+shift+f` combination, the cursor is jumping to the EOF.

**Comment by VanguardaSistemas** [comment]
Same issue here

**Comment by ValentinLeTallec** [comment]
Same here, except that with

- a ~5000 lines file 
- vscode 1.77.3
- prettier 9.10.4

marekcambal's workaround did not work for me

**Comment by dnjat** [comment]
+1 crazy

**Comment by dnjat** [comment]
bug +1. this is bug,no enhancement

**Comment by liboz** [comment]
I was able to determine that the bug was introduced in 9.5. Revert to version 9.4 fixes the issue. Looking at the commit log, https://github.com/prettier/prettier-vscode/commit/be790ca13b85b26196340ce2a89a1f8a47545605 seems very suspect. 

I changed the mode locally to just modifications but then prettier doesn't run at all. I'm guessing that the algorithm to get only the modified part is too costly to run for very large files.

The solution is probably to only use this algorithm when the setting is enabled.

**Comment by rotu** [comment]
> I was able to determine that the bug was introduced in 9.5. Reverting to version 9.4 fixes the issue. Looking at the commit log, [be790ca](https://github.com/prettier/prettier-vscode/commit/be790ca13b85b26196340ce2a89a1f8a47545605) seems very suspect.

I don't remember exactly why my change allows range formatting to work. I think it was because prettier always reports that it changed the entire file and this would overlap with some other edit ranges from somewhere else. VSCode would see the conflicting edits and bail.

Because of [@madc0w's analysis](https://github.com/prettier/prettier-vscode/issues/2920#issuecomment-1481289885), I searched for "100000", in VSCode, and this looks like a contributing factor: https://github.com/microsoft/vscode/blob/f125afbc800ec611f5a9ab1333c769832ce424b3/src/vs/editor/common/services/editorSimpleWorker.ts#L492

Looking at the blame, that line was last changed to fix [vscode#72321](https://github.com/microsoft/vscode/issues/72321) and the discussion there seems *very* relevant.

**Comment by 38b3eff8** [comment]
+1

**Comment by TheSecurityDev** [comment]
Just came across this issue after the file got larger. Previously it was working fine but now this particular file always jumps to the bottom. I have prettier configured to format on save, so saving causes this issue, which is extremely annoying since I save often.

**Comment by Bahman-Lugano** [comment]
Any resolution on this issue? It's very annoying and counterproductive. Please fix this as it seems to be a straight forward and easy fix especially since the older versions of prettier don't seem to have the same issue.

**Comment by obinecko** [comment]
well looks like still nothing

**Comment by Danielku15** [comment]
I narrowed down the problem for my project! 🥳  

**tl;dr:** Its a problem with inconsistent line endings between the editor and prettier defaults. Ensure consistent line endings in prettier and VSCode. 

Would be great if affected people here can confirm my assumptions for their environment. 

**deeper insights**

My setup:

* `.prettierrc` - No setting (defaults to LF) 
* I am on Windows with auto crlf on git level 
* I have a editorconfig which would force LF
* I have no gitattributes which would force LF, hence git applies auto CRLF and my files are with CRLF
* vscode auto-detects the line endings from the content and sets the editor to CRLF
* When formatting I see in the output window: `"endOfLine": "lf"`

As soon I change the settings to be consistent (in VSCode or in prettier) the jumping disappears. I guess one leads to another and messes up the offsets:

* Prettier delivers `\n` lines for the partial edit
* Prettier vscode tries to calculate the correct text range for the optimized edit but its not very optimized anymore:
   * only the first line of the file until the line ending matches leading almost to a full file replace. 
* The file is replaced almost fully with the `\n` string. 
* From what I see VSCode replaces again the line endings with CRLF.
* Somewhere in VSCode the cursor gets messed up. 

Unless I find another spot where `prettier-vscode` is to blame, we need to carry this to [microsoft/vscode](https://github.com/microsoft/vscode).

1. Prettier and prettier-vscode are doing things correctly. As per expectation they replace the content to be with LF instead of CRLF. 
2. prettier-vscode could try to change the editor mode to match the configured setting. This should eliminate the problem on subsequent formats. (https://github.com/prettier/prettier-vscode/blob/b8d9e42b8900c88e2210bc72615d9f925439b058/src/PrettierEditService.ts#L134 -> `editBuilder.setEndOfLine`
3. prettier-vscode could warn users about the inconsistency in their project setup. (refer to https://prettier.io/docs/options#end-of-line)

For (2) I gave it locally a quick shot to implement it. As you can see it jumps on the first formatting and it updates the editor to to LF. After that the formatting seem to act normally. 

https://github.com/user-attachments/assets/059b568e-f522-43e8-ac50-a0327ddef8a4

**Comment by thxSakura** [comment]
"prettier.endOfLine": "auto" 
fix for me

**Comment by milkyway0308** [comment]
> "prettier.endOfLine": "auto" fix for me

It works to me too. You are my savior!

**Comment by MMMMoriaty** [comment]
> "prettier.endOfLine": "auto" fix for me
thx bro
